From 5880cab67e4b3f6ab40fd6f3477bd16f6eb329cf
From: "Lucie L. Hartmann" <lucie@mntre.com>
Subject: WORKAROUND: meson_plane: disable AFBC 32x8 support
Date:   Sat Mar 22 14:48:13 2025 +0100

    Since mesa/panfrost has turned on 32x8 AFBC compression for Mali-G52 we are getting missed frames/syncs on our A311D MNT Reforms and Pocket Reforms. While the problem is investigated, we turn off this compression format in the display pipeline for this SoC, so we don't have to patch mesa.

    Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1099032
    Mesa bug: https://gitlab.freedesktop.org/mesa/mesa/-/issues/12718

--- a/drivers/gpu/drm/meson/meson_plane.c
+++ b/drivers/gpu/drm/meson/meson_plane.c
@@ -502,25 +502,6 @@ static const uint64_t format_modifiers_a
 	 * - 32x8 block size + SPLIT is mandatory with 4K frame size
 	 *   for performances reasons
 	 */
-	DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_16x16 |
-				AFBC_FORMAT_MOD_SPARSE |
-				AFBC_FORMAT_MOD_SPLIT),
-	DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_16x16 |
-				AFBC_FORMAT_MOD_YTR |
-				AFBC_FORMAT_MOD_SPARSE |
-				AFBC_FORMAT_MOD_SPLIT),
-	DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_32x8 |
-				AFBC_FORMAT_MOD_SPARSE),
-	DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_32x8 |
-				AFBC_FORMAT_MOD_YTR |
-				AFBC_FORMAT_MOD_SPARSE),
-	DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_32x8 |
-				AFBC_FORMAT_MOD_SPARSE |
-				AFBC_FORMAT_MOD_SPLIT),
-	DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_32x8 |
-				AFBC_FORMAT_MOD_YTR |
-				AFBC_FORMAT_MOD_SPARSE |
-				AFBC_FORMAT_MOD_SPLIT),
 	DRM_FORMAT_MOD_LINEAR,
 	DRM_FORMAT_MOD_INVALID,
 };
